### PR TITLE
DOC: 'MS' offset alias does not work as described with pd.date_range()

### LINKED ISSUE #44146: DOC: 'MS' offset alias does not work as described with pd.date_range()
### 

- [X] I have checked that the issue still exists on the latest versions of the docs on `master` [here](https://pandas.pydata.org/docs/dev/)

### Location of the documentation

https://pandas.pydata.org/pandas-docs/stable/user_guide/timeseries.html#offset-aliases

### Documentation problem

The current documentation for the `MS` offset alias reads as follows: "month start frequency", indicating that it is the complement of `M` (which is "month end frequency").

However, for any date supplied to `pd.date_range(start=[date], ... freq='MS')` other than the first of the month, this offset alias starts from the beginning of the **next** month. See [this](https://stackoverflow.com/questions/34915828/pandas-date-range-to-generate-monthly-data-at-beginning-of-the-month) StackOverflow post (second answer) for a more detailed explanation. 

However, in brief:
```
start = "2020-03-08"
end = "2021-03-08"
pd.date_range(start, end, freq='MS')
```
gives
```
DatetimeIndex(['2020-04-01', '2020-05-01', '2020-06-01', '2020-07-01',
           '2020-08-01', '2020-09-01', '2020-10-01', '2020-11-01',
           '2020-12-01', '2021-01-01', '2021-02-01', '2021-03-01'],
          dtype='datetime64[ns]', freq='MS')
```

It should be noted that supplying the first day of the month as the start date keeps the expected behaviour (i.e. the start point of the range is the first of the month supplied).

### Suggested fix for documentation

If the behaviour of the `MS` offset alias is indeed correct (i.e. to start at the beginning of the **next** month), then this should be indicated in the documentation with a note.

For example:
"""
`MS` = month start frequency. Note, for any day other than the first day of the month this will cause the offset to start at the beginning of the next month.
"""

The above explanation correctly encapsulates the current functionality of the `MS` offset, especially within `pd.date_range()`. The current documentation would incorrectly lead a person to believe that the `MS` offset would cause the start date to be the start of the month of the date supplied.

If this behaviour is not correct, and `MS` should be causing the start date to be the start of the month of the date supplied, then a subsequent bug would need to be raised.

### PR DESCRIPTION
**Pull request by loicdiridollou**
- [x] closes #44146 
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [NA] whatsnew entry

### CODE REVIEW

File: doc/source/user_guide/timeseries.rst [hunk 1]
```diff
@@ -1271,6 +1271,21 @@ frequencies. We will refer to these aliases as *offset aliases*.
     "U, us", "microseconds"
     "N", "nanoseconds"
 
+.. note::
+
```
Comment (jreback) [inline doc/source/user_guide/timeseries.rst:1275]: umm w/o explicit examples I am not sure how useful this note is, so pls add these

File: doc/source/user_guide/timeseries.rst [hunk 2]
```diff
@@ -1271,6 +1271,21 @@ frequencies. We will refer to these aliases as *offset aliases*.
...
+   When using the offset aliases above, it should be noted that functions
+   such as ``date_range``, ``bdate_range``, will only return timestamps that
```
Comment (jreback) [inline doc/source/user_guide/timeseries.rst:1277]: also if you are adding here it is ok, but would be great to add to the actual functions as well

Comment (jreback) [reply doc/source/user_guide/timeseries.rst:1277]: use something like ``` :func:`pd.date_range` ``` to reference this

**Review by jreback — CHANGES_REQUESTED** [review]

File: doc/source/user_guide/timeseries.rst (hunk 2 above)
Comment (loicdiridollou) [reply doc/source/user_guide/timeseries.rst:1277]: Correct sorry I missed this :func:, this is updated

**Comment by jreback** [comment]
pls build the docs locally and see where this is going wrong: https://github.com/pandas-dev/pandas/runs/4280860460?check_suite_focus=true

**Comment by loicdiridollou** [comment]
> pls build the docs locally and see where this is going wrong: https://github.com/pandas-dev/pandas/runs/4280860460?check_suite_focus=true

My mistake I misaligned some text that got included in the ipython block, will be more careful next time, should be fixed now (but waiting for full build)

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @loicdiridollou

**Pull request merged**